src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -119,30 +129,35 @@ def accept_bid(self, bid: Bid, energy: float = None,
             changed_bid = Bid(
                 str(uuid.uuid4()), residual_price, residual_energy,
                 buyer, seller,
-                original_bid_price=energy_portion * orig_price
+                original_bid_price=(1 - energy_portion) * orig_price
```

Comment:
This change is not due to the new implementation of the grid fees, right? This was a bug, right?

Reply:
You are right, this is a bug that has been lingering there for quite a while, and I wonder why did we not find it until now.